### PR TITLE
Add support for juliaup

### PR DESCRIPTION
**Pull request by rmolina**
## Standards checklist:

- [*] The PR title is descriptive.
- [*] The code compiles (`cargo build`)
- [*] The code passes rustfmt (`cargo fmt`)
- [*] The code passes clippy (`cargo clippy`)
- [*] The code passes tests (`cargo test`)
- [*] *Optional:* I have tested the code myself
    - [ ] I also tested that Topgrade skips the step where needed

This adds support for juliaup. It's just a copy of rustup.

### CODE REVIEW

**Comment by guihkx** [comment]
Can you please set the base branch to `dev` instead of `master`, and then rebase your PR against it? Thank you.

File: src/config.rs [hunk 1]
```diff
@@ -127,6 +127,7 @@ pub enum Step {
     Restarts,
     Rtcl,
     Rustup,
+    Juliaup,
```
Comment (guihkx) [inline src/config.rs:130]: nit: These are sorted alphabetically, please move it up.

Comment (rmolina) [reply src/config.rs]: Done. Thank you.

**Comment by rmolina** [comment]
> Can you please set the base branch to `dev` instead of `master`, and then rebase your PR against it? Thank you.

Done. Thank you.

File: src/steps/generic.rs [hunk 2]
```diff
@@ -175,6 +175,18 @@ pub fn run_rustup(base_dirs: &BaseDirs, run_type: RunType) -> Result<()> {
...
+pub fn run_juliaup(base_dirs: &BaseDirs, run_type: RunType) -> Result<()> {
+    let juliaup = utils::require("juliaup")?;
+
+    print_separator("juliaup");
+
+    if juliaup.canonicalize()?.is_descendant_of(base_dirs.home_dir()) {
+        run_type.execute(&juliaup).args(["self", "update"]).check_run()?;
+    }
+
+    run_type.execute(&juliaup).arg("update").check_run()
```
Comment (DottoDev) [inline src/steps/generic.rs:187]: Could you please replace chech_run with status_checked(). Ex.: https://github.com/topgrade-rs/topgrade/blob/c83c850546d00c2d73b47263ed6c622e72463afe/src/steps/generic.rs#L175

**Review by DottoDev — CHANGES_REQUESTED** [review]

File: src/steps/generic.rs (hunk 2 above)
Comment (rmolina) [reply src/steps/generic.rs]: Done. Thanks.

**Review by DottoDev — APPROVED** [review]

**Comment by DottoDev** [comment]
Looks good. Will be merged after the 10.2.0 release.

**Pull request merged**